### DISCUSSION TITLE
the choice of constant flow rate and constant pressure gradient

### DISCUSSION THREAD

**Discussion by jushi-123** [xcompact3d/Incompact3d#313] [Q&A] [answered]
Hello,
I want to ask that in the simulation of turbulent channel flow ,wheather the constant flow rate is realized by adjusting the velocity to match the bulk velocity and the constant pressure gradient is realized by adding a force(constant) in the momentum equation directly?When I use the constant flow rate, whether the friction reynolds number will be different of that in CPG?
Many thanks

**Comment by slaizet** [comment, marked as answer]
Hi,
For the turbulent channel flow, the constant flow rate is obtained by adjusting the velocity to match the bulk velocity.
For the pressure gradient forcing, the Reynolds number in the input file is the targeted Re_tau (when using the constant flow rate forcing, the Reynolds number in the input is based on the bulk velocity. 

Both methods lead to the same statistics but the normalisation on the input file is different.

I hope it makes sense.

Best,

Sylvain

**Reply by jushi-123** [reply]
So when I use the constant flow rate to simulate the turbulent channel flow, the Reynolds number in input file is based on bulk velocity(Re_bulk=U_bulk*H/viscosity), instead of the Reynolds number which is based on the centerline velocity of laminar Poiseuille flow?

**Reply by slaizet** [reply]
The input Reynolds number is Based on the centreline velocity of a Poiseuille flow.
For Re_theta=180, Re_input=Re_centreline=4200, Re_bulk=2800 (basically Re_bulk=(2/3) Re_centreline). We use Re_centreline because we normalise the simulation with U_centreline = 1.
Best,
Sylvain

**Reply by jushi-123** [reply]
Thank you very much! According to you suggestion, I think the error in my code maybe some other reasons instead of Reynolds number setting.

**Reply by jushi-123** [reply]
Hello Dr Laizet,
I have some doubts in the constant flow rate of turbulent channel flow, wheather the velocity at the wall should be adjusted to match the bulk velocity at each time step?According to the user guide of Incompact3d, the subroutine pre_correc is used to apply the no-slip boundary condition. The following is the subroutine used in present program.
[subroutine channel_cfr.txt](https://github.com/user-attachments/files/17977839/subroutine.channel_cfr.txt)

**Reply by slaizet** [reply]
No problem here, as the forcing term for the constant flow rate, even if applied at the wall is overwritten by the boundary conditions at the wall in the subroutine pre_correc.
Best,
Sylvain
